cuda/base/config.hpp

```diff
@@ -74,6 +74,21 @@ struct config {
 };
 
 
+template <typename ValueType>
+struct batch_config {
+    /**
+     * Max number of rows per matrix in a batch of (small) sparse matrices.
+     */
+    static constexpr int max_num_rows = 128;
+
+    /**
+     * Max number of RHS vectors in a linear system. This can be at most
+     * `config::warp_size`.
+     */
+    static constexpr int max_num_rhs = 5;
```

Comment:
Is there any reason, you set this to 5, rather than `config::warp_size` ?

Reply:
The static allocation depended on this. Setting it to 32 would make us run out shared memory much quicker for no good reason. If we choose to permanently remove static allocation, this whole struct can be removed. Since this is not user-facing, we could also do this later.

Reply:
Yeah, we can decide this later.